language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/SnippetBlock.java

```diff
@@ -55,6 +57,16 @@ public SnippetBlock(String label, String snippet, String detail, Kind kind) {
         this.detail = detail;
         this.kind = kind;
         this.imports = null;
+        this.additionalTextEdits = null;
+    }
+
+    public SnippetBlock(String label, String snippet, String detail, Kind kind, List<TextEdit> additionalTextEdits) {
```

Comment:
Shall we avoid modifying the SnippetBlock? This is because the SnippetBlock is supposed to be only for the hard-coded snippets such as the main function.